clouddriver-dcos/src/main/groovy/com/netflix/spinnaker/clouddriver/dcos/provider/config/DcosProviderConfig.groovy

```diff
@@ -0,0 +1,100 @@
+package com.netflix.spinnaker.clouddriver.dcos.provider.config
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.fasterxml.jackson.databind.SerializationFeature
+import com.netflix.spectator.api.Registry
+import com.netflix.spinnaker.cats.agent.Agent
+import com.netflix.spinnaker.cats.provider.ProviderSynchronizerTypeWrapper
+import com.netflix.spinnaker.clouddriver.dcos.DcosCloudProvider
+import com.netflix.spinnaker.clouddriver.dcos.provider.DcosProvider
+import com.netflix.spinnaker.clouddriver.dcos.security.DcosAccountCredentials
+import com.netflix.spinnaker.clouddriver.dcos.security.DcosClusterCredentials
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsProvider
+import com.netflix.spinnaker.clouddriver.security.AccountCredentialsRepository
+import com.netflix.spinnaker.clouddriver.security.ProviderUtils
+import org.springframework.beans.factory.config.ConfigurableBeanFactory
+import org.springframework.context.annotation.Bean
+import org.springframework.context.annotation.Configuration
+import org.springframework.context.annotation.DependsOn
+import org.springframework.context.annotation.Scope
+
+import java.util.concurrent.ConcurrentHashMap
+
+@Configuration
+class DcosProviderConfig {
+  @Bean
+  @DependsOn('dcosCredentials')
+  DcosProvider dcosProvider(DcosCloudProvider dcosCloudProvider,
+                            AccountCredentialsProvider accountCredentialsProvider,
+                            AccountCredentialsRepository accountCredentialsRepository,
+                            ObjectMapper objectMapper,
+                            Registry registry) {
+
+    def provider = new DcosProvider(dcosCloudProvider, Collections.newSetFromMap(new ConcurrentHashMap<Agent, Boolean>()))
+    synchronizeDcosProvider(provider, accountCredentialsProvider, accountCredentialsRepository, objectMapper, registry)
+    provider
+  }
+
+  @Bean
+  DcosProviderSynchronizerTypeWrapper dcosProviderSynchronizerTypeWrapper() {
+    new DcosProviderSynchronizerTypeWrapper()
+  }
+
+  // TODO this doesnt appear to be getting used, since we're not running a scheduled synchronizer
```

Comment:
Implementing this allows for the set of credentials to be updated without cycling clouddriver. But that codepath is essentially unused by anyone that we know of (you would have to POST to a particular endpoint to exercise it).